python/ray/serve/_private/client.py

```diff
@@ -479,8 +481,8 @@ def get_handle(
             if cached_handle._is_same_loop:
                 return cached_handle
 
-        all_endpoints = ray.get(self._controller.get_all_endpoints.remote())
-        if not missing_ok and deployment_name not in all_endpoints:
+        all_deployments = self.list_deployments()
```

Comment:
Actually I think since this is calling `client.list_deployments()` instead of the public API, there is no deserialization happening. However since we only need the deployment names anyways, I've changed the implementation to call an API on the controller to only fetch the names. I also added a unit test like you described below!

Reply:
Nice, I didn't catch that this was calling `client.list_deployments()` instead of the public API. Thanks for adding the test!